src/tail/mod.rs

```diff
@@ -52,28 +53,19 @@ impl Tail {
             let tunnel = tokio::spawn(tunnel_process.run(tunnel_rx));
 
             // Register the tail with the Workers API and send periodic heartbeats
-            let session = tokio::spawn(Session::run(target, user, session_rx));
+            let session = tokio::spawn(Session::run(target, user, session_rx, tx));
 
-            let res = tokio::try_join!(listener, log_server, session, tunnel);
+            let res = tokio::try_join!(
+                async { listener.await? },
+                async { log_server.await? },
+                async { session.await? },
+                async { tunnel.await? }
+            );
```

Comment:
wwwtttttffffff

Reply:
yeah so this is effectively the thing i was talking about in the tokio-users discord where i was told that actually those join handles created by `tokio::spawn` wrap the future in another result/future, so this pulls them out. it's bonkers and imho try_join documentation should be clearer about this, or the macro should do this for you, but perhaps there is some utility to this approach. either way it didn't make sense for me to remove the parallelism provided by spawning new tasks, so this is what happened. probably deserves a comment though.